src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -132,54 +137,63 @@ class MonitorMetadata(SubjectMetadata):
 def configure(config):
     """Maps commands/events to their functions."""
     middleware = [
-        command_context_middleware,
-        db_middleware,
+        subject_middleware,
         configuration_middleware,
     ]
 
     # don't need an incident context
-    app.command(config.slack_command_list_incidents, middleware=[db_middleware])(
-        handle_list_incidents_command
+    middleware.extend([non_incident_command_middlware])
+
+    app.command(config.slack_command_report_incident, middleware=middleware)(
+        ack=ack_command, lazy=[handle_report_incident_command]
     )
-    app.command(config.slack_command_report_incident, middleware=[db_middleware])(
-        handle_report_incident_command
+    app.command(config.slack_command_list_incidents, middleware=middleware)(
+        ack=ack_command, lazy=[handle_list_incidents_command]
     )
 
     # non-sensitive-commands
-    app.command(config.slack_command_list_tasks, middleware=middleware)(handle_list_tasks_command)
+    middleware.remove(non_incident_command_middlware)
+    middleware.extend([command_context_middleware])
```

Comment:
This is a bit confusing, should we build a new middleware list instead of modifying it?

Reply:
Agreed, made this change, and I think it's more simple.